Initial value transfer app

jwbambacht opened the pull request:
The value transfer app combines peerchat, eurotoken, and ssi attestations into one useable and deployable application. Many features are implemented, although some may have to be improved. Some features do not function very well at the moment, but will be changed in a later version.
To use this app the user is required to fill in its identity details. Using this identity it is then possible to communicate with other identities. The public key of the users' identity is equal to the public key used in the superapp itself.

jwbambacht commented:
@synctext I have made a PR but cannot get the errors fixed. I don't think these errors are caused by the code that I've written. I also saw that in the PR of sjacobino the same errors are shown. Is there someone that can take a look at it?

xoriole commented:
@jwbambacht It is failing at `./gradlew check` action with linting issues.
>  \> Lint found errors in the project; aborting build.

Running it locally, I see that there are some linting issues on `valuetransfer` and `peerchat` packages as well. Could you fix them and update the PR?

Since Ktlint is used to enforce a consistent code style across the whole project,
you can run the lint check with:
```
./gradlew ktlintCheck
```
and format the code with:
```
./gradlew ktlintFormat
```

jwbambacht commented:
@xoriole Thanks for your recommendation. I have fixed the lint issues locally and updated the PR. The checks still fail, giving the same error in kotlin-ipv8 in which I have changed nothing. Since I do not have much experience with this I have no clue what to do with that.

jwbambacht commented:
@xoriole it indeed fixed the permissions issues and passed all tests

xoriole commented on valuetransfer/src/test/java/nl/tudelft/trustchain/valuetransfer/ExampleUnitTest.kt:
```diff
@@ -0,0 +1,17 @@
+package nl.tudelft.trustchain.valuetransfer
```
this file can be removed, does not add any value

xoriole commented on valuetransfer/src/main/java/nl/tudelft/trustchain/valuetransfer/ui/contacts/ContactChatFragment.kt:
```diff
@@ -0,0 +1,882 @@
+package nl.tudelft.trustchain.valuetransfer.ui.contacts
```
please do a clean up of the code removing commented out code.

xoriole commented on valuetransfer/src/main/java/nl/tudelft/trustchain/valuetransfer/ui/contacts/ContactChatFragment.kt:
```diff
@@ -0,0 +1,882 @@
+            }
+        }
+
+        lifecycleScope.launchWhenCreated {
```
this block of code seems redundant since most of the code is commented out

xoriole commented on valuetransfer/src/androidTest/java/nl/tudelft/trustchain/valuetransfer/ExampleInstrumentedTest.kt:
```diff
@@ -0,0 +1,24 @@
+package nl.tudelft.trustchain.valuetransfer
```
this example test file can be removed, does not add any value

xoriole commented on app/build.gradle:
```diff
@@ -45,6 +46,19 @@ android {
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
     }
 
+    // To run a different version next to standard version of the app uncomment flavorDimensions and productFlavors, comment 'client' in google-services.json, and select demo as build variant
```
Please remove these commented out lines about the flavors and line changes related to flavors

xoriole reviewed (changes requested):
@jwbambacht Could you please process these change requests?

jwbambacht commented:
@xoriole requested changes are done and all checks passed
